index.js

```diff
@@ -98,59 +98,190 @@ export default function ({
 				// If the root was not pre-rendered, add a placeholder index.html
 				// Route all requests for the index to the SSR function
 				writeFileSync(`${staticDir}/index.html`, '');
-				swaConfig.routes.push(
-					{
-						route: '/index.html',
-						rewrite: ssrFunctionRoute
-					},
-					{
-						route: '/',
-						rewrite: ssrFunctionRoute
-					}
-				);
 			}
 
+			const swaConfig = generateConfig(builder, customStaticWebAppConfig);
+
 			writeFileSync(`${publish}/staticwebapp.config.json`, JSON.stringify(swaConfig));
 		}
 	};
+	return adapter;
 }
 
 /**
- * @param {import('./types/swa').CustomStaticWebAppConfig} customStaticWebAppConfig
- * @param {string} appDir
+ * @param {import('@sveltejs/kit').Builder} builder A reference to the SvelteKit builder
+ * @param {import('./types/swa').StaticWebAppConfig} customStaticWebAppConfig Custom configuration
  * @returns {import('./types/swa').StaticWebAppConfig}
  */
-export function generateConfig(customStaticWebAppConfig, appDir) {
-	validateCustomConfig(customStaticWebAppConfig);
-
-	if (!customStaticWebAppConfig.routes) {
-		customStaticWebAppConfig.routes = [];
-	}
-
+export function generateConfig(builder, customStaticWebAppConfig = {}) {
+	builder.log.minor(`Generating Configuration (staticwebapp.config.json)...`);
 	/** @type {import('./types/swa').StaticWebAppConfig} */
 	const swaConfig = {
 		...customStaticWebAppConfig,
-		routes: [
-			...customStaticWebAppConfig.routes,
-			{
-				route: '*',
-				methods: ['POST', 'PUT', 'DELETE'],
-				rewrite: ssrFunctionRoute
-			},
-			{
-				route: `/${appDir}/immutable/*`,
-				headers: {
-					'cache-control': 'public, immutable, max-age=31536000'
-				}
-			}
-		],
 		navigationFallback: {
-			rewrite: ssrFunctionRoute
+			rewrite: ssrFunctionRoute,
+			...customStaticWebAppConfig.navigationFallback
 		},
 		platform: {
-			apiRuntime: 'node:16'
-		}
+			apiRuntime: nodeVersion,
+			...customStaticWebAppConfig.platform
+		},
+		routes: []
 	};
 
+	if (swaConfig.navigationFallback.rewrite === ssrTrigger) {
+		swaConfig.navigationFallback.rewrite = ssrFunctionRoute;
+	}
```

Comment:
There are a few use cases when it would not break the site and would be preferable because it prevents usage charges from bots all over the world scanning your site for vulnerabilities at paths that don't exist. 

1. For a fully static site, there is no need for SSR and it makes sense to disable it entirely.
2. For a site with explicit endpoints, the routes with SSR enabled can be narrowly defined.

And, of course, if the configuration key is not set it cannot break the adapter. It just gives more options for projects that use the adapter.

Reply:
For 1, the user of the adapter shouldn't need to do anything for that - you configure [prerendered pages in SvelteKit](https://kit.svelte.dev/docs/page-options#prerender) and Azure SWA will automatically route to those prerendered pages without hitting the SSR function.

For 2, I'd want to see an example of what the use case is there. `navigationFallback` is meant to be the final fallback if nothing else resolves the GET request. If you make it anything other than the generated `render` function, SvelteKit's client-side routing will break completely.

Reply:
> It just gives more options for projects that use the adapter.

I want to avoid exposing an option unless there is a specific use-case to solve for. More options/features means more things to maintain and consider when making other updates to the adapter.

Reply:
The specific use-case is to prevent 404 errors from racking up Azure function usage charges on routes that are never supposed to be SSR. Yes, you can configure prerendered pages in SvelteKit, but 404 errors will be rendered by an Azure function with this adapter as it is currently built. 

> I want to avoid exposing an option unless there is a specific use-case to solve for. More options/features means more things to maintain and consider when making other updates to the adapter.

I agree generally, but this code only gets the adapter out of the way of features provided by Azure. It allows the custom configuration to work the way Microsoft's documentation describes, and it has no impact on people who don't use it.